.github/workflows/ci.yml

```diff
@@ -20,3 +20,16 @@ jobs:
           coverage: xdebug
       - run: composer install
       - run: vendor/bin/phpunit --coverage-text
+
+  static-analysis:
+    name: PHPStan
+    runs-on: ubuntu-20.04
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.1
+      - run: composer require phpstan/phpstan
```

Comment:
Should this be added to `require-dev` so it's easier to run this inside the project directory?

Reply:
Yes, it should. I copied this from promise where due to version constraints on the v2 branch it's more complicated.